thavalon-webapp/src/utils/AccountManager.tsx

```diff
@@ -73,48 +90,44 @@ export class AccountManager {
             "message": "",
         }
         // returns 500 or 401 or 200
-        return await fetch("/api/auth/refresh", {
+        const response: Response = await fetch("/api/auth/refresh", {
             method: "POST",
             credentials: "include"
-        }).then((response) => {
-            switch(response.status) {
-                case STATUS.OK: {
-                    // this.setJwtInfo.bind(this);
-                    response.json().then((jwt: JwtType) => {
-                        // use anonymous function to get around this being unbound
-                        this.setJwtInfo(jwt);
-                    });
-                    break;
-                }
-                case STATUS.UNAUTHORIZED: {
-                    httpResponse.result = false;
-                    httpResponse.message = "Unauthorized when trying to refresh token";
-                    break;
-                }
-                case STATUS.INTERNAL_SERVER_ERROR: {
-                    httpResponse.result = false;
-                    httpResponse.message = "Internal server error when refreshing token";    
-                    break;
-                }
-                default: {
-                    httpResponse.result = false;
-                    httpResponse.message = "Unexpected return code from server: " + response.status;    
-                    break;
-                }
-            }
-            // log any non-OK statuses, and clear jwt info
-            if (!httpResponse.result) {
-                console.log("Invalid refresh token, Reason: " + httpResponse.message);
-                this.token = "";
-                this.expiresAt = 0;        
-            }    
-            return httpResponse;
-        }).catch((error) => {
-            console.log("Failed to refresh token, error is: " + error);
-            httpResponse.result = false;
-            httpResponse.message = "Unable to refresh token, try again";
-            return httpResponse;
         });
+
+        switch(response.status) {
+            case STATUS.OK: {
+                // this.setJwtInfo.bind(this);
+                response.json().then((jwt: JwtType) => {
```

Comment:
Up to you, but this could also be an await

Reply:
oops missed it here, will change